### DISCUSSION TITLE
How pass headers to custom activity

### DISCUSSION THREAD

**Discussion by tuohaibei** [elsa-workflows/elsa-core#4558] [Q&A] [answered]
I have one workflow,but i want pass headers from http endpoint to start working activity,is there any solution?
![image](https://github.com/elsa-workflows/elsa-core/assets/5082315/597854b5-8882-4ea0-b2cd-c328d6076210)

**Comment by sfmskywalker** [comment, marked as answer]
Yes, you can access the HTPT headers sent to the HTTP Endpoint activity from other activities, such as your custom activity.
Not knowing anything about your custom activity, let me give you an example with the following workflow:

<img width="433" alt="image" src="https://github.com/elsa-workflows/elsa-core/assets/938393/061baf68-9cc6-4337-8dc0-b5cea1790061">

That workflow listens at /workflows/hello-world. The HTTP Endpoint activity has the following name: **MyHttpEndpoint**. This is important, because we are going to reference this activity from the HTTP Response activity in a second.

Now, consider the following HTTP request:

```
curl --location 'https://localhost:11000/workflows/hello-world' \
--header 'x-test: Hello Header!' \
--header 'Content-Type: application/json' \
--data '{
}'
```

Notice that I am sending a header called `x-test`.
If we want to access the value of that header in the HTTP Response activity, we can do that like this:


```javascript
activities.MyHttpEndpoint.Output().Headers["x-test"]
```

When invoking the cURL, you will receive a response that echoes the value back that was sent via the x-test header.

I [attached the workflow](https://github.com/elsa-workflows/elsa-core/files/13043978/workflow-definition-7dbc59e0c0244473af28ea22141459be.json.zip) in case you want to take a closer look.

**Comment by tuohaibei** [comment]
Thank you,I have solved the problem the way you said.
